data/mods/ssb/moves.js

```diff
@@ -991,6 +991,36 @@ let BattleMovedex = {
 		target: "self",
 		type: "Psychic",
 	},
+	// c.kilgannon
+	insidiousassault: {
+		accuracy: 100,
+		basePower: 70,
+		category: "Special",
+		desc: "This move's category becomes physical if the opponent's Defense stat is lower than its Special Defense stat. This move's Base Power is doubled if this move is physical.",
+		shortDesc: "Physical and power doubles if foe's Def < SpD.",
+		id: "insidiousassault",
+		name: "Insidious Assault",
+		isNonstandard: "Custom",
+		pp: 5,
+		priority: 0,
+		flags: {protect: 1, mirror: 1},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Black Hole Eclipse', target);
+		},
+		onModifyMove(move, pokemon, target) {
+			if (target.getStat('def', false, true) > target.getStat('spd', false, true)) move.category = 'Physical';
+		},
+		onBasePower(basePower, source, target, move) {
+			if (move.category === 'Physical') {
+				return this.chainModify(2);
```

Comment:
Really should be modifying the foe's defense before hitting not doubling BP. There should be an event for this.